specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/assessments.json

```diff
@@ -0,0 +1,330 @@
+            }
+          }
+        }
+      },
+      "put": {
+        "x-ms-examples": {
+          "Create security recommendation task on a resource": {
+            "$ref": "./examples/Assessments/PutAssessment_example.json"
+          }
+        },
+        "tags": [
+          "Assessments"
+        ],
+        "description": "Create a security assessment on your resource. An assessment metadata that describes this assessment must be predefined with the same name before inserting the assessment result",
+        "operationId": "Assessments_Create",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceId"
+          },
+          {
+            "$ref": "#/parameters/AssessmentName"
+          },
+          {
+            "$ref": "#/parameters/AssessmentBody"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SecurityAssessment"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SecurityAssessmentList": {
+      "type": "object",
+      "description": "List of security assessments",
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SecurityAssessment"
+          }
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The URI to fetch the next page."
+        }
+      }
+    },
+    "SecurityAssessment": {
+      "type": "object",
+      "description": "Security assessment on a resource",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SecurityAssessmentProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "SecurityAssessmentProperties": {
+      "type": "object",
+      "description": "Describes properties of an assessment.",
+      "properties": {
+        "resourceDetails": {
+          "$ref": "../../../common/v1/types.json#/definitions/ResourceDetails"
+        },
+        "displayName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "User friendly display name of the assessment"
+        },
+        "status": {
+          "$ref": "#/definitions/AssessmentStatus"
+        },
+        "additionalData": {
+          "type": "object",
```

Comment:
What is additionalData? Can it be modeled more explicitly than object?

Reply:
every assessment have its own details to add here
unfortunately we cannot model this more explicitly

Reply:
every assessment have its own details to add here
unfortunately we cannot model this more explicitly